lib/ansible/plugins/inventory/toml.py

```diff
@@ -211,8 +212,8 @@ def _load_file(self, file_name):
             raise AnsibleFileNotFound("Unable to retrieve file contents", file_name=file_name)
 
         try:
-            with open(b_file_name, 'r') as f:
-                return toml.load(f)
+            (b_data, private) = self.loader._get_file_contents(file_name)
+            return toml.loads(b_data.decode("utf-8"))
```

Comment:
```suggestion
            return toml.loads(to_text(b_data, errors='surrogate_or_strict'))
```

This will also require adding `to_text` to the `from ansible.module_utils._text import to_bytes, to_native` line.

Reply:
done.